[Bug][connector][starrocks] write starRocks failed.

### Search before asking

- [X] I had searched in the [issues](https://github.com/DTStack/chunjun/issues) and found no similar issues.


### What happened

当从es -> starrocks 同步时报错, es 有一千多万条数据，发现sr每次写入的时候都要去校验表机构，这块校验每次都要去查sr元数据，每次写入都校验是否有必要，能否优化？

```

2023-06-27 16:30:12,259 INFO  org.apache.flink.runtime.executiongraph.ExecutionGraph       [] - Source: source[3] -> Calc[4] -> ConstraintEnforcer[5] -> Sink: sink[5] (1/1) (2e9745297f837566e6ee0b4d71c91d70_6cdc5bb954874d922eaee11a8e7b5dd5_0_0) switched from RUNNING to FAILED on container_e04_1686304823865_0531_01_000002 @ xxxx (dataPort=46282).
com.dtstack.chunjun.throwable.ChunJunRuntimeException: write starRocks failed.
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeRecordInternal(StarRocksOutputFormat.java:109) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.sink.format.BaseRichOutputFormat.lambda$initTimingSubmitTask$0(BaseRichOutputFormat.java:465) ~[chunjun-core.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_144]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_144]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_144]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_144]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_144]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_144]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_144]
Caused by: java.lang.Exception: Writing records to StarRocks failed.
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.write(StreamLoadManager.java:91) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.NormalWriteProcessor.write(NormalWriteProcessor.java:57) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeMultipleRecordsInternal(StarRocksOutputFormat.java:70) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeRecordInternal(StarRocksOutputFormat.java:97) ~[chunjun-connector-starrocks-master.jar:?]
	... 8 more
Caused by: java.lang.RuntimeException: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Cannot open null host.
	at com.dtstack.chunjun.util.RetryUtil$Retry.doRetry(RetryUtil.java:125) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.util.RetryUtil.executeWithRetry(RetryUtil.java:47) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.connector.starrocks.connection.StarRocksJdbcConnectionProvider.getConnection(StarRocksJdbcConnectionProvider.java:67) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.connection.StarRocksJdbcConnectionProvider.checkValid(StarRocksJdbcConnectionProvider.java:84) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StarRocksQueryVisitor.executeQuery(StarRocksQueryVisitor.java:103) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StarRocksQueryVisitor.getTableColumnsMetaData(StarRocksQueryVisitor.java:69) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StarRocksQueryVisitor.getTableColumnsMetaData(StarRocksQueryVisitor.java:58) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.validateTableStructure(StreamLoadManager.java:99) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.lambda$write$0(StreamLoadManager.java:74) ~[chunjun-connector-starrocks-master.jar:?]
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[?:1.8.0_144]
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.write(StreamLoadManager.java:65) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.NormalWriteProcessor.write(NormalWriteProcessor.java:57) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeMultipleRecordsInternal(StarRocksOutputFormat.java:70) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeRecordInternal(StarRocksOutputFormat.java:97) ~[chunjun-connector-starrocks-master.jar:?]
	... 8 more
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Cannot open null host.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:403) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.Util.getInstance(Util.java:386) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:944) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3933) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3869) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2524) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2675) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2465) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2439) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1365) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.loadServerVariables(ConnectionImpl.java:3767) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.initializePropsFromServer(ConnectionImpl.java:3196) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2233) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2015) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:768) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47) ~[chunjun-connector-mysql-master.jar:?]
	at sun.reflect.GeneratedConstructorAccessor32.newInstance(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:403) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:385) ~[chunjun-connector-mysql-master.jar:?]
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:323) ~[chunjun-connector-mysql-master.jar:?]
	at java.sql.DriverManager.getConnection(DriverManager.java:664) ~[?:1.8.0_144]
	at java.sql.DriverManager.getConnection(DriverManager.java:208) ~[?:1.8.0_144]
	at com.dtstack.chunjun.connector.starrocks.connection.StarRocksJdbcConnectionProvider.lambda$getConnection$0(StarRocksJdbcConnectionProvider.java:69) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.util.RetryUtil$Retry.call(RetryUtil.java:129) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.util.RetryUtil$Retry.doRetry(RetryUtil.java:71) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.util.RetryUtil.executeWithRetry(RetryUtil.java:47) ~[chunjun-core.jar:?]
	at com.dtstack.chunjun.connector.starrocks.connection.StarRocksJdbcConnectionProvider.getConnection(StarRocksJdbcConnectionProvider.java:67) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.connection.StarRocksJdbcConnectionProvider.checkValid(StarRocksJdbcConnectionProvider.java:84) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StarRocksQueryVisitor.executeQuery(StarRocksQueryVisitor.java:103) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StarRocksQueryVisitor.getTableColumnsMetaData(StarRocksQueryVisitor.java:69) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StarRocksQueryVisitor.getTableColumnsMetaData(StarRocksQueryVisitor.java:58) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.validateTableStructure(StreamLoadManager.java:99) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.lambda$write$0(StreamLoadManager.java:74) ~[chunjun-connector-starrocks-master.jar:?]
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[?:1.8.0_144]
	at com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager.write(StreamLoadManager.java:65) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.NormalWriteProcessor.write(NormalWriteProcessor.java:57) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeMultipleRecordsInternal(StarRocksOutputFormat.java:70) ~[chunjun-connector-starrocks-master.jar:?]
	at com.dtstack.chunjun.connector.starrocks.sink.StarRocksOutputFormat.writeRecordInternal(StarRocksOutputFormat.java:97) ~[chunjun-connector-starrocks-master.jar:?]
	... 8 more
2023-06-27 16:30:12,281 INFO  org.apache.flink.runtime.resourcemanager.slotmanager.DeclarativeSlotManager [] - Clearing resource requirements of job 32d5c958cceb02fbdfdaf8c20b08a0b8

```

### What you expected to happen

--

### How to reproduce



```

CREATE TABLE source
(
  `agent` string,
  `route_id` string,
  `scheme` string,
  `request_completion` string,
  `consumer_name` string,
  `request_time` float,
  `proxy_host` string,
  `host` string,
  `user_ip` string,
  `method` string,
  `prophet_token` string,
  `uri` string,
  `args` string,
  `@timestamp` string,
  `request_id` string,
  `status` string
) WITH (
      'password' = 'xxxx',
      'connector' = 'elasticsearch7-x',
      'sink.bulk-flush.max-actions' = '10000',
      'hosts' = 'xxxx',
      'client.connect-timeout' = '10000',
      'index' = 'xxxx',
      'search-query' = '{

    "range":{
     
      "@timestamp": {"gte":"2023-05-01T10:00:00.000Z",
                                 "lt":"2023-05-20T00:00:00.000Z"}
  
  }
  
}',
      'username' = 'xxx'
      );
CREATE TABLE sink
(
  `agent` string,
  `route_id` string,
  `scheme` string,
  `request_completion` string,
  `consumer_name` string,
  `request_time` string,
  `proxy_host` string,
  `host` string,
  `user_ip` string,
  `method` string,
  `prophet_token` string,
  `uri` string,
  `args` string,
  `referer` string,
  `request_id` string,
  `status` string,
  PRIMARY KEY (referer) NOT ENFORCED
) WITH (
      'password' = '0xxxx',
      'connector' = 'starrocks-x',
      'schema-name' = 'xxxx',
      'table-name' = 'xxxx',
      'fe-nodes' = 'xxxx',
      'url' = 'jdbc:mysql://xxxxx/xxxx?useUnicode=true&characterEncoding=utf-8&useSSL=false&connectTimeout=3000&useUnicode=true&characterEncoding=utf8&useSSL=false&rewriteBatchedStatements=true&&serverTimezone=Asia/Shanghai',
      'username' = 'xxxx'
      );
insert into sink 
select 
substring(agent,1,200) as `agent`,
`route_id` as `route_id`,
`scheme` as `scheme`,
`request_completion` as `request_completion`,
`consumer_name` as `consumer_name`,
CAST(`request_time` AS string) as `request_time`,
`proxy_host` as `proxy_host`,
`host` as `host`,
`user_ip` as `user_ip`,
`method` as `method`,
`prophet_token` as `prophet_token`,
substring(uri,1,200) as `uri`,
substring(args,1,200) as `args`,
`@timestamp` as `referer`,
`request_id` as `request_id`,
`status` as `status`
 from source;

```




### Anything else

_No response_

### Version

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/chunjun-connectors/chunjun-connector-starrocks/pom.xml
+++ b/chunjun-connectors/chunjun-connector-starrocks/pom.xml
@@ -59,6 +59,12 @@
 			<artifactId>arrow-memory-netty</artifactId>
 			<version>${arrow.version}</version>
 		</dependency>
+
+		<dependency>
+			<groupId>com.alibaba</groupId>
+			<artifactId>fastjson</artifactId>
+			<version>${fastjson.version}</version>
+		</dependency>
 	</dependencies>
 
 	<build>

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/config/StarRocksConfig.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/config/StarRocksConfig.java
@@ -53,6 +53,18 @@ public class StarRocksConfig extends CommonConfig {
     private Integer maxRetries = 3;
 
     // sink
+    /**
+     * Whether to save the table structure information obtained from the source for each
+     * tableIdentify
+     */
+    private boolean isCacheTableStruct = true;
+
+    /**
+     * In a multi-table write scenario, whether to check the table structure only once for each
+     * table.This parameter is valid only when isCacheTableStruct=true
+     */
+    private boolean checkStructFirstTime = true;
+
     /** The time to sleep when the tablet version is too large */
     private long waitRetryMills = 18000;
 

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/options/StarRocksSinkOptions.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/options/StarRocksSinkOptions.java
@@ -23,6 +23,20 @@ import org.apache.flink.configuration.ConfigOptions;
 
 public class StarRocksSinkOptions {
 
+    public static final ConfigOption<Boolean> SINK_CACHE_TABLE_STRUCT =
+            ConfigOptions.key("cache_table_struct")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription(
+                            "Whether to save the table structure information obtained from the source for each tableIdentify");
+
+    public static final ConfigOption<Boolean> SINK_CHECK_STRUCT_FIRST_TIME =
+            ConfigOptions.key("check_struct_first_time")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription(
+                            "In a multi-table write scenario, whether to check the table structure only once for each table.This parameter is valid only when CACHE_TABLE_STRUCT=true");
+
     public static final ConfigOption<String> SINK_SEMANTIC =
             ConfigOptions.key("semantic")
                     .stringType()

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/MappedWriteProcessor.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/MappedWriteProcessor.java
@@ -72,7 +72,7 @@ public class MappedWriteProcessor extends StarRocksWriteProcessor {
             valueList.add(value);
         }
         for (Map.Entry<String, List<Map<String, Object>>> entry : identifyXValueMap.entrySet()) {
-            streamLoadManager.write(entry.getKey(), null, entry.getValue());
+            streamLoadManager.write(entry.getKey(), null, entry.getValue(), true);
         }
     }
 }

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/NormalWriteProcessor.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/NormalWriteProcessor.java
@@ -19,6 +19,7 @@
 package com.dtstack.chunjun.connector.starrocks.sink;
 
 import com.dtstack.chunjun.connector.starrocks.config.StarRocksConfig;
+import com.dtstack.chunjun.connector.starrocks.streamload.StarRocksSinkBufferEntity;
 import com.dtstack.chunjun.connector.starrocks.streamload.StreamLoadManager;
 import com.dtstack.chunjun.converter.AbstractRowConverter;
 
@@ -44,6 +45,10 @@ public class NormalWriteProcessor extends StarRocksWriteProcessor {
         super(streamLoadManager, starRocksConfig);
         this.rowConverter = converter;
         this.columnNameList = columnNameList;
+
+        streamLoadManager.validateTableStructure(
+                new StarRocksSinkBufferEntity(
+                        starRocksConfig.getDatabase(), starRocksConfig.getTable(), columnNameList));
     }
 
     @Override
@@ -54,6 +59,6 @@ public class NormalWriteProcessor extends StarRocksWriteProcessor {
         for (RowData rowData : rowDataList) {
             values.add(rowConverter.toExternal(rowData, new HashMap<>(columnNameList.size())));
         }
-        streamLoadManager.write(identify, columnNameList, values);
+        streamLoadManager.write(identify, columnNameList, values, false);
     }
 }

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/StarRocksDynamicTableSink.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/StarRocksDynamicTableSink.java
@@ -70,6 +70,7 @@ public class StarRocksDynamicTableSink implements DynamicTableSink {
                         Arrays.asList(resolvedSchema.getColumnNames().toArray(new String[0]))));
         builder.setStarRocksConf(sinkConfig);
 
+        builder.checkFormat();
         return SinkFunctionProvider.of(
                 new DtOutputFormatSinkFunction<>(builder.finish()), sinkConfig.getParallelism());
     }

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/StarRocksOutputFormatBuilder.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/sink/StarRocksOutputFormatBuilder.java
@@ -46,6 +46,9 @@ public class StarRocksOutputFormatBuilder
                     "starRocks database is required when nameMapped is not enable");
             Preconditions.checkNotNull(
                     conf.getTable(), "starRocks table is required when nameMapped is not enable");
+            // cache is not necessary because we only have one table
+            conf.setCacheTableStruct(false);
+            conf.setCheckStructFirstTime(false);
         }
         Preconditions.checkNotNull(conf.getUsername(), "starRocks username is required");
         Preconditions.checkNotNull(conf.getPassword(), "starRocks password is required");

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/streamload/StreamLoadManager.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/streamload/StreamLoadManager.java
@@ -20,6 +20,8 @@ package com.dtstack.chunjun.connector.starrocks.streamload;
 
 import com.dtstack.chunjun.connector.starrocks.config.StarRocksConfig;
 
+import org.apache.flink.util.Preconditions;
+
 import com.alibaba.fastjson.JSON;
 import lombok.extern.slf4j.Slf4j;
 
@@ -38,6 +40,7 @@ import java.util.concurrent.TimeUnit;
 public class StreamLoadManager {
     final LinkedBlockingDeque<StarRocksSinkBufferEntity> flushQueue = new LinkedBlockingDeque<>(1);
     private final Map<String, StarRocksSinkBufferEntity> bufferMap = new ConcurrentHashMap<>();
+    private final Map<String, List<Map<String, Object>>> structCacheMap = new ConcurrentHashMap<>();
 
     private final StarRocksConfig starRocksConfig;
     private final boolean __opAutoProjectionInJson;
@@ -57,7 +60,11 @@ public class StreamLoadManager {
         this.starrocksStreamLoadVisitor = new StarRocksStreamLoadVisitor(starRocksConfig);
     }
 
-    public void write(String tableIdentify, List<String> columnList, List<Map<String, Object>> data)
+    public void write(
+            String tableIdentify,
+            List<String> columnList,
+            List<Map<String, Object>> data,
+            boolean checkTableStructure)
             throws Exception {
         try {
             checkFlushException();
@@ -71,7 +78,9 @@ public class StreamLoadManager {
                                                 databaseAndTable[0],
                                                 databaseAndTable[1],
                                                 columnList);
-                                validateTableStructure(starRocksSinkBufferEntity);
+                                if (checkTableStructure) {
+                                    validateTableStructure(starRocksSinkBufferEntity);
+                                }
                                 return starRocksSinkBufferEntity;
                             });
             bufferEntity.addToBuffer(
@@ -92,14 +101,44 @@ public class StreamLoadManager {
         }
     }
 
+    private List<Map<String, Object>> getTableStruct(StarRocksSinkBufferEntity entity) {
+        if (starRocksConfig.isCacheTableStruct()) {
+            return structCacheMap.computeIfAbsent(
+                    getStructCacheKey(entity),
+                    k -> getTableStructFromSource(entity.getDatabase(), entity.getTable()));
+        } else {
+            return getTableStructFromSource(entity.getDatabase(), entity.getTable());
+        }
+    }
+
+    private String getStructCacheKey(StarRocksSinkBufferEntity entity) {
+        return entity.getDatabase() + "_" + entity.getTable();
+    }
+
+    private List<Map<String, Object>> getTableStructFromSource(String database, String table) {
+        return starrocksQueryVisitor.getTableColumnsMetaData(database, table);
+    }
+
     public void validateTableStructure(StarRocksSinkBufferEntity entity) {
         if (starRocksConfig.getLoadConfig().getHeadProperties().containsKey("columns")) {
             return;
         }
-        List<Map<String, Object>> rows = starrocksQueryVisitor.getTableColumnsMetaData();
-        if (rows == null || rows.isEmpty()) {
-            throw new IllegalArgumentException("Couldn't get the sink table's column info.");
+        if (starRocksConfig.isCheckStructFirstTime()
+                && starRocksConfig.isCacheTableStruct()
+                && structCacheMap.containsKey(getStructCacheKey(entity))) {
+            return;
         }
+        List<Map<String, Object>> rows =
+                Preconditions.checkNotNull(
+                        getTableStruct(entity),
+                        String.format(
+                                "Couldn't get the sink table[%s] column info.",
+                                getStructCacheKey(entity)));
+        validateTableStructure(rows, entity);
+    }
+
+    public void validateTableStructure(
+            List<Map<String, Object>> rows, StarRocksSinkBufferEntity entity) {
         // validate primary keys
         List<String> primaryKeyList = new ArrayList<>();
         Set<String> containedColumnNameSet = new HashSet<>();

--- a/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/table/StarRocksDynamicTableFactory.java
+++ b/chunjun-connectors/chunjun-connector-starrocks/src/main/java/com/dtstack/chunjun/connector/starrocks/table/StarRocksDynamicTableFactory.java
@@ -48,6 +48,8 @@ import static com.dtstack.chunjun.connector.starrocks.options.StarRocksCommonOpt
 import static com.dtstack.chunjun.connector.starrocks.options.StarRocksCommonOptions.USERNAME;
 import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSinkOptions.NAME_MAPPED;
 import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSinkOptions.SINK_BUFFER_FLUSH_MAX_ROWS;
+import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSinkOptions.SINK_CACHE_TABLE_STRUCT;
+import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSinkOptions.SINK_CHECK_STRUCT_FIRST_TIME;
 import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSinkOptions.SINK_SEMANTIC;
 import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSourceOptions.FILTER_STATEMENT;
 import static com.dtstack.chunjun.connector.starrocks.options.StarRocksSourceOptions.SCAN_BE_CLIENT_KEEP_LIVE_MIN;
@@ -139,11 +141,15 @@ public class StarRocksDynamicTableFactory
     private StarRocksConfig createSinkConfByOptions(ReadableConfig options) {
         StarRocksConfig sinkConf = createCommonConfByOptions(options);
         // sink options
+        boolean cacheTableStruct = options.get(SINK_CACHE_TABLE_STRUCT);
+        boolean checkStructFirstTime = options.get(SINK_CHECK_STRUCT_FIRST_TIME);
         boolean nameMapped = options.get(NAME_MAPPED);
         Integer batchSize = options.get(SINK_BUFFER_FLUSH_MAX_ROWS);
         Long sinkInternal = options.get(SINK_BUFFER_FLUSH_INTERVAL);
         LoadConfig loadConfig = getLoadConfig(options);
         // loading
+        sinkConf.setCacheTableStruct(cacheTableStruct);
+        sinkConf.setCheckStructFirstTime(checkStructFirstTime);
         sinkConf.setNameMapped(nameMapped);
         sinkConf.setBatchSize(batchSize);
         sinkConf.setFlushIntervalMills(sinkInternal);
@@ -224,6 +230,8 @@ public class StarRocksDynamicTableFactory
 
         // sink
         optionalOptions.add(NAME_MAPPED);
+        optionalOptions.add(SINK_CACHE_TABLE_STRUCT);
+        optionalOptions.add(SINK_CHECK_STRUCT_FIRST_TIME);
         optionalOptions.add(SINK_BUFFER_FLUSH_MAX_ROWS);
         optionalOptions.add(SINK_BUFFER_FLUSH_INTERVAL);
         optionalOptions.add(SINK_SEMANTIC);

